kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/serialization/KotlinSerializer.kt

```diff
@@ -0,0 +1,205 @@
+ * The serialized format is JSON.
+ *
+ * The DSL function kotlinSerializer can be used to easily configure the parameters
+ * for this serializer.
+ *
+ * @see kotlinx.serialization.Serializer
+ * @see org.axonframework.serialization.Serializer
+ * @see kotlinSerializer
+ */
+class KotlinSerializer(
+    private val revisionResolver: RevisionResolver,
+    private val converter: Converter,
+    private val json: Json
+) : Serializer {
+
+    private val serializerCache: MutableMap<Class<*>, KSerializer<*>> = mutableMapOf()
+
+    override fun <T> serialize(value: Any?, expectedRepresentation: Class<T>): SerializedObject<T> {
+        try {
+            val type = ObjectUtils.nullSafeTypeOf(value)
+            return when {
+                expectedRepresentation.isAssignableFrom(String::class.java) ->
+                    SimpleSerializedObject(
+                        (if (value == null) "null" else json.encodeToString(type.serializer(), value)) as T,
+                        expectedRepresentation,
+                        typeForClass(type)
+                    )
+
+                expectedRepresentation.isAssignableFrom(JsonElement::class.java) ->
+                    SimpleSerializedObject(
+                        (if (value == null) JsonNull else json.encodeToJsonElement(type.serializer(), value)) as T,
+                        expectedRepresentation,
+                        typeForClass(type)
+                    )
+
+                else ->
+                    throw org.axonframework.serialization.SerializationException("Cannot serialize type $type to representation $expectedRepresentation. String and JsonElement are supported.")
+            }
+        } catch (ex: SerializationException) {
+            throw org.axonframework.serialization.SerializationException("Cannot serialize type ${value?.javaClass?.name} to representation $expectedRepresentation.", ex)
+        }
+    }
+
+    override fun <T> canSerializeTo(expectedRepresentation: Class<T>): Boolean =
+        expectedRepresentation == String::class.java ||
+                expectedRepresentation == JsonElement::class.java
+
+    override fun <S, T> deserialize(serializedObject: SerializedObject<S>?): T? {
+        try {
+            if (serializedObject == null) {
+                return null
+            }
+
+            if (serializedObject.type == SerializedType.emptyType()) {
+                return null
+            }
+
+            val serializer: KSerializer<T> = serializedObject.serializer()
+            return when {
+                serializedObject.contentType.isAssignableFrom(String::class.java) ->
+                    json.decodeFromString(serializer, serializedObject.data as String)
+
+                serializedObject.contentType.isAssignableFrom(JsonElement::class.java) ->
+                    json.decodeFromJsonElement(serializer, serializedObject.data as JsonElement)
+
+                else ->
```

Comment:
Just as with the `serialize` method, I miss the `byte[]` option here.

Reply:
:+1: